lambdaclass/champions_of_mirra
# Kaline tree level up

Closes #374 

⚠️ Test using this branch on the backend: https://github.com/lambdaclass/mirra_backend/pull/499

## Summary of changes

Implement the level up of the Kaline Tree, as well as the display of the leveling costs in the Kaline Tree scene.

## How has this been tested?

1. Start the game and go to the Kaline Tree.
2. Initially, you have enough Fertilizer and Gold to level up the tree once, so you should be able to do it.
3. After leveling up, check that gold and fertilizer were consumed by running `GameBackend.Users.Currencies.get_user_currency(user_id, GameBackend.Users.Currencies.get_currency_by_name!("Gold").id)` and `GameBackend.Users.Currencies.get_user_currency(user_id, GameBackend.Users.Currencies.get_currency_by_name!("Fertilizer").id)` with the respective user_id.
4. Make sure that the gold in the header was updated, and that the leveling costs increased. As we haven't decided a formula for upgrading yet, leveling costs are equal to `100 * kaline_tree_level`.
5. Try to level up once more and verify that there's an error shown in a pop up.

## Checklist
